### DISCUSSION TITLE
Mutability of Grid object

### DISCUSSION THREAD

**Discussion by anissa111** [UXARRAY/uxarray#108] [Q&A] [answered]
Hello uxarray-ers!

I have a question for y'all. Is there ever a use case that we'd like to support where a user could change some of the variables set in grid initialization? Is this something we even should be supporting?

For example, if a user wants to add additional points to make an area of a grid finer, should we allow the original grid object to be modified?

@brianpm, interested to hear your thoughts if you have any!

**Comment by brianpm** [comment]
I can't really think of a use case for this, but maybe @paullric or others know better. In all my use cases, at least that I can imagine right now, some other software has created a grid and I'm just working with data on that grid. The exception would be if this kind of mutability is used for remapping between grids ... I could definitely imagine use cases for that. For example, if I had data on an unstructured mesh and I just wanted to extract a geographic region on a regular mesh (without having to generate a "destination grid" a priori). That might be totally separate from this issue though.

**Comment by paullric** [comment, marked as answer]
Almost always when you want to do an operation on a Grid you'd generate a new Grid, since it would require extensive modification of the Grid data (e.g., computing a grid dual).  However, you may want to add more information to a Grid object, such as face areas, but that shouldn't affect the mutability of the vertex/face/edge data.
